modules/core/src/main/scala/gem/enum/package.scala

```diff
@@ -30,4 +34,38 @@ package object enum {
         case SmartGcalType.NightBaseline => baseline(GcalBaselineType.Night)
       }
   }
+
+  @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+  implicit class HalfCompanionOps(companion: Half.type) {
+
+    def unsafeFromInt(n: Int): Half =
```

Comment:
`Half` is generated from the db along with all the other enumerated types. Look in `modules/core/target/scala-2.12/src_managed/main/gem`.

The `.toInt` method gives a numeric meaning to the ordering, which lets us do math with semesters by treating `(year, half)` as a single value `2y + h`, which turns `plusSemesters/prev/next` into simple integer arithmetic without the weird boundary logic. It could be inlined in `Semester` but it seemed like it might come in handy elsewhere so I pushed it down into `Half`.